pgmpy/independencies/Independencies.py

```diff
@@ -422,17 +422,25 @@ def __init__(self, event1=[], event2=[], event3=[]):
         self.all_vars = frozenset().union(self.event1, self.event2, self.event3)
 
     def __str__(self):
+        def _format_event(event):
+            if all(isinstance(item, tuple) for item in event):
+                # Dynamic case
+                return ", ".join(f'{item}' for item in event)
+            else:
+                # Static case
+                return ", ".join(event)
+
         if self.event3:
             return "({event1} _|_ {event2} | {event3})".format(
-                event1=", ".join(self.event1),
-                event2=", ".join(self.event2),
-                event3=", ".join(self.event3),
+                event1=_format_event(self.event1),
+                event2=_format_event(self.event2),
+                event3=_format_event(self.event3),
```

Comment:
@nihil21 I don't think this is the best way from a maintenance point of view. I think the function would get confusing later on, which would make maintaining it difficult. I would instead keep it simple and just convert everything to str. Something like: `", ".join([str(event) for event in self.event1])`